UserTools/MeanTimeCheck/MeanTimeCheck.cpp

```diff
@@ -0,0 +1,185 @@
+#include "MeanTimeCheck.h"
+
+MeanTimeCheck::MeanTimeCheck():Tool(){}
+
+
+bool MeanTimeCheck::Initialise(std::string configfile, DataModel &data){
+
+  /////////////////// Useful header ///////////////////////
+  if(configfile!="") m_variables.Initialise(configfile); // loading config file
+  //m_variables.Print();
+
+	std::string output_filename;
+	m_variables.Get("verbosity", verbosity);
+	m_variables.Get("OutputFile", output_filename);
+	m_variables.Get("ShowEvent", ShowEvent);
+	Output_tfile = new TFile(output_filename.c_str(), "recreate");
+
+	delta = new TH1D("delta", "delta", 1000, -10, 30);
+	peakTime = new TH1D("Peak Time", "time", 1000, -10, 30);
+	basicAverage = new TH1D("basic average", "time", 1000, -10, 30);
+	weightedPeak = new TH1D("Weighted Peak", "time", 1000, -10, 30);
+	weightedAverage = new TH1D("Weighted Average", "time", 1000, -10, 30);
+
+  m_data= &data; //assigning transient data pointer
+  /////////////////////////////////////////////////////////////////
+
+  return true;
+}
+
+
+bool MeanTimeCheck::Execute(){
+
+Log("===========================================================================================", v_debug, verbosity);
+
+	Log("MeanTimeCheck Tool: Executing", v_debug, verbosity);
+	auto* reco_event = m_data->Stores["RecoEvent"];
+	if (!reco_event) {
+		Log("Error: The MeanTimeCheck tool could not find the ANNIEEvent Store", 0, verbosity);
```

Comment:
again name of store doesn't tie up with the error message, and this isn't a good way to do this. Same comments as DirectionGridCheck.